src/meta/meta_ckpt.c

```diff
@@ -613,6 +623,16 @@ __ckpt_load(WT_SESSION_IMPL *session, WT_CONFIG_ITEM *k, WT_CONFIG_ITEM *v, WT_C
     WT_RET_NOTFOUND_OK(ret);
     if (ret != WT_NOTFOUND && a.len != 0)
         ckpt->ta.newest_stop_durable_ts = (uint64_t)a.val;
+    else {
+        /*
+         * Backward compatibility changes, as the parameter name is different in older versions of
+         * WT, make sure that we read older format in case if we didn't find the newer format name.
+         */
+        ret = __wt_config_subgets(session, v, "stop_durable_ts", &a);
+        WT_RET_NOTFOUND_OK(ret);
+        if (ret != WT_NOTFOUND && a.len != 0)
+            ckpt->ta.newest_stop_durable_ts = (uint64_t)a.val;
```

Comment:
Maybe assert that it's `WT_TS_NONE`?

Reply:
No. we cannot add an assert here, because when the database is upgrading from 4.4rc3 to later, there can be a value that is not `WT_TS_NONE`. But when it is upgrading from 4.2 data files, there is no `stop_durable_ts` in the metadata and it just returns NOTFOUND, so the default value `WT_TS_NONE` is initialized.